From bc90e8679f9c6bf8af33125586da213cab5dcc59 Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Thu, 29 Jun 2017 14:30:16 +0200
Subject: [PATCH 2/3] Fix use-after-free in initialize_gio()

Don't use already freed use_vfs variable in the fuse code, but use the
disable_fuse one.

https://bugzilla.gnome.org/show_bug.cgi?id=784339
---
 gnome-session/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnome-session/main.c b/gnome-session/main.c
index 89e8f3b5..d28f0df9 100644
--- a/gnome-session/main.c
+++ b/gnome-session/main.c
@@ -258,7 +258,7 @@ initialize_gio (void)
         }
 
         if (disable_fuse) {
-                g_setenv ("GVFS_DISABLE_FUSE", use_vfs, TRUE);
+                g_setenv ("GVFS_DISABLE_FUSE", disable_fuse, TRUE);
                 g_free (disable_fuse);
         } else {
                 g_unsetenv ("GVFS_DISABLE_FUSE");
-- 
2.13.0

